capstone/src/main/java/com/step902020/capstone/UserController.java

```diff
@@ -0,0 +1,81 @@
+
+package com.step902020.capstone;
+
+import java.io.IOException;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+import com.google.gson.Gson;
+import java.io.IOException;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@RestController
+public class UserController {
+
+  @Autowired
+  private IndividualRepository individualRepository;
+
+  @Autowired
+  private OrganizationRepository organizationRepository;
+
+  @GetMapping("get-individual")
+  public List<Individual> getIndividual(@RequestParam("email") String email) {
+    return this.individualRepository.findByEmail(email);
+  }
+
+  @GetMapping("get-organization")
+  public List<Organization> getOrganization(@RequestParam("email") String email) {
+    return this.organizationRepository.findByEmail(email);
+  }
+
+  @PostMapping("save-individual")
+  public RedirectView saveIndividual(
+      @RequestParam("id") String id,
+      @RequestParam("firstname") String firstname,
+      @RequestParam("lastname") String lastname, 
+      @RequestParam("email") String email, 
+      @RequestParam("user-type") String userType,
+      @RequestParam("university") String university) throws IOException {
+
+    // each email can only exist as either an individual or a user, not both
+    this.organizationRepository.deleteByEmail(email);
```

Comment:
Non-actionable for now, since this is test data: in real life, this would enable a disgruntled student deleting an organization's profile by creating a personal profile with the same e-mail address.

Reply:
Actually this is something I was struggling with. Since we are storing individuals and users separately I need to check for duplicate accounts in the other user type. I also need to fetch from both entities. I was wondering if there's some way around it? Like query both entities at the same time through a common property?

Reply:
It might be easier to keep organizations and individuals more separate. For instance, you can have a separate controller for organizations, and even separate forms/pages for management.

Reply:
They are different entity types, so you can't query them in one go. But querying both and failing with a meaningful message if you find an existing entity of a different type is better than silently deleting the conflicting entry.